ompi/instance/instance.c

```diff
@@ -964,6 +965,9 @@ static void ompi_instance_get_num_psets_complete (pmix_status_t status,
             if (ompi_mpi_instance_pmix_psets) {
                 opal_argv_free (ompi_mpi_instance_pmix_psets);
             }
+            if (NULL != pset_names) {
+                free(pset_names);
```

Comment:
Can't this lead to a potential double free?  Below this, there's a case where you `goto done`, which also calls `free(pset_names)`.

If de-snaggling that logic is too difficult, you could add a `pset_names = NULL` here after the `free()`, and that would make the 2nd `free(pset_names)` safe.

Reply:
@jsquyres  check now